rholang/src/test/scala/coop/rchain/rholang/ProcGen.scala

```diff
@@ -20,7 +21,7 @@ object tools {
 
   def streamSingleton[T](v: T): Stream[T] = v #:: Stream.empty[T]
 
-  def nonemptyString(g: Gen[Char], size: Int): Gen[String] = Gen.nonEmptyListOf(g).map(_.mkString)
+  def nonemptyString(size: Int): Gen[String] = Gen.alphaStr.suchThat(_.size > 0)
```

Comment:
Size is unused. We should probably wrap it in `Gen.sized(size) { }`

Reply:
What do you mean that `size` is unused? It is but I think the implementation can be even more efficient.

Reply:
How it is now:

`def nonemptyString(size: Int): Gen[String] = Gen.alphaStr.map(_.take(size))`

is wasteful (generates more than is needed), and potentially acts against what the user wants if they pass the size obtained from the Arbitrary. It really should say:

`Gen.resize(size, alphaStr.suchThat(_.size > 0))`

Sorry for mixing `resize` with `sized` in my prev comment :)